### ISSUE TITLE
Camera toggle takes long time and freezes picture

### ISSUE THREAD

**Issue by akovalov** [shogo4405/lf.swift#175]
Hello! Here is some weird behavior.
I have a separate ViewController for streaming, I show it, camera is loaded and streaming all good. Then I tap button to toggle camera once or multiple times, in code
```
cameraPosition = cameraPosition == .back ? .front : .back
rtmpStream.attachCamera(DeviceUtil.device(withPosition: cameraPosition))
```
and it could change fast and no problem but in some cases could take 10 s to switch cameras blocking main thread UI, or can switch in 1-2 s but with frozen mirrored last frame from previous camera and not showing real image, switching back to previous camera and it shows real picture again. 

I've tried to debug and figured out that time delay is in two places when doing addOutput and removeOutput
```
func attachCamera(_ camera:AVCaptureDevice?) { 
   output = nil // seter for it does mixer.session.removeOutput(output) which is taking time
...
   mixer.session.addOutput(output) // takes time
...
```
According to Apple Doc https://developer.apple.com/reference/avfoundation/avcapturesession 

> The startRunning() method is a blocking call which can take some time, therefore you should perform session setup on a serial queue so that the main queue isn't blocked (which keeps the UI responsive). See AVCam-iOS: Using AVFoundation to Capture Images and Movies for an implementation example.

And I wonder why it freezes picture sometimes?..
Do you have any ideas on toggle time delays and picture freezes?

Thanks.

**Comment by shogo4405** [comment]
Related issue #104. I think fixed 0.4.2 or 0.5.2...
Could you tell me, iOS version, device, framework version?

Regards.

**Comment by akovalov** [comment]
Framework version is 0.5.8, iOS 10.2, iPhone 6.
Ok, I think I figured it out. So I have separate view controller to stream in app, and RTMPStreamController which is created inside vc and I see in Allocations instrument that after close vc object and controller get deallocated and not lives anymore, however all lf classes continue to live, so when vc is shown again all lf classes created again after so 2 instances of each are live. 
I've changed so my RTMPStreamController is singleton and looks like delay and freeze issues not appear. 
I'm not sure if I capture strong references somewhere to keep objects or it's somewhere inside framework. Does framework intended to work as single instance only and not created / removed during app session?

**Comment by shogo4405** [comment]
> Does framework intended to work as single instance only and not created / removed during app session?

No, framework doesn't. I will try to fix.

**Comment by akovalov** [comment]
Thanks!

**Comment by shogo4405** [comment]
fixed this issue :) see also #177
I will release 0.5.11 in a few days.

**Comment by shogo4405** [comment]
Released 0.5.11
